### ISSUE TITLE
runDemo

### ISSUE THREAD

**Issue by hadley** [rstudio/shiny#212]
It would be useful if shiny had a `runDemo` function like this:

``` R
runDemo <- function(package, demo, ...) {
  path <- system.file("demo", demo, package = package)
  if (path == "") stop("Couldn't find demo ", demo, " in ", package, call. = FALSE)

  runApp(path, ...)
}
```

then you could easily run demos in installed packages:

``` R
runDemo("apps/basic", "gigvis")
```

That might almost be a better way to implement `runExample()`:

``` R
runExample <- function(example, ...) runApp("shiny", example, ...)
```

although then you'd have to move the existing examples into the demo directory - this would be a good idea because it's fits the R package conventions a little better, and reduces the confusion between `runExample()` and `example()`

**Comment by dougmet** [comment]
Found this just before posting a similar issue. I think it would be nice to add a `package` argument to `runExample`. This would turn the following line:

```r
shiny::runApp(system.file("shiny-examples", "basic", package="radarchart"))
```
into
```r
shiny::runExample("basic", package="radarchart")
```

Happy to submit a PR if there's interest. The main thing is to agree on a standard directory to keep shiny examples. Here I've used `shiny-examples`, similar to [this post](http://deanattali.com/2015/04/21/r-package-shiny-app/) but some use `examples` some use `demo`.

For me the name isn't important, but having this function as part of shiny would be a great way to encourage consistency.
